### DISCUSSION TITLE
[Feature Request] Change to how "Your Completion Progress" are displayed

### DISCUSSION THREAD

**Discussion by rayden-61** [RetroAchievements/RAWeb#2441] [Ideas]
![image](https://github.com/RetroAchievements/RAWeb/assets/6521658/ba50b2ef-a91d-4092-b567-dd811774a7e2)

Hi, I'd like to suggest a change to how "Your Completion Progress" numbers are organized. Currently, you separate Softcore and Hardcore numbers as the following fields. I will use some values as examples:

- Beaten (softcore): 11
- Beaten: 13
- Completed (only counts softcore completions): 1
- Mastered: 2

My suggestion is to replace softcore entries with a total between both softcore + hardcore, and keep the hardcore entries as they are. 
So, using the same numbers, it would become like this:

- Beaten (Total): 24
- Beaten (Hardcore): 13
- Completed (Total): 3
- Mastered (Hardcore) 2

There are some advantages to my suggestion. The first obvious one is that bigger number = better. Users get more engaged when they have a bigger stat to care about instead of spreading their progression numbers across multiple stats. Also, a "total number of games you've beaten" is certainly a much more interesting statistic to know at a glance than "total number of games you've beaten minus the ones you didn't use savestates????". And even if you do care about knowing the exact number of softcore completions, it's just a matter of subtracting total from hardcore.

I don't think anyone would be opposed to this way of displaying things, but if that is controversial, I guess a setting on the user's preferences to how they want to see these metrics would be enough to please everyone. Like " Display completion numbers: separate vs. combined"

**Comment by wescopeland** [comment]
I generally like this idea. There are a few nuances to consider:
* We want to be judicious about the creation of new user settings. For the time being, I'd like to consider this off the table.
* I believe this change would also need to logically be applied to the "Progression Status" component that appears on user profiles (the thing that lists the consoles and their beaten/mastered status).

I would be interested in getting a sense from more users who have mixed progress how they might feel about this proposed change. @Hexadigital do you have an opinion?

**Comment by Hexadigital** [comment]
The main thing I dislike about this proposal is that it overemphasizes hardcore, but if the subcategories changed between hardcore and softcore depending on which the user had more of, I think it could be fine?

The other aspect that would be odd is if a player solely plays hardcore or softcore - in that case, you'd either have duplicated values (where total and the game mode are the same number) or ambiguous descriptions (where it shows just the total, but doesn't state if they are hardcore or softcore).

If I were to make a proposal, I'd suggest to just show the total number for each category (played, unfinished, beaten, completed) regardless of game mode - since a game that is mastered is a game that was completed, and the Status filter can be used to distinguish/see the two modes if needed.

**Reply by rayden-61** [reply]
I don't see that my suggestion overemphasizes hardcore necessarily. The thing is: softcore completion is just a generic type of completion, while hardcore completion has more specific requirements. In my mind, if you beat or complete a game on hardcore, you should also, for all intents and purposes, have it count as having achieved all of the requirements for a softcore completion as well. It feels counterintuitive when the site counts both separately as if they have different specific requirements, when in actuality, hardcore is just "all of the softcore requirements, but more".

There's another scenario in which I believe my suggestion would provide a good experience for users: say you have started your retroachievements journey only playing softcore and have lots of softcore completions. The more softcore completions you have, the less incentive there is to switch to hardcore, because your "number of completed games" will, in a sense, start from scratch. Say you have like... 90 completions, Instead of continuing your progress towards 100, switching to hardcore at that point will net you an ugly "90 of this one thing and 1 of that other thing I guess". I know it sounds nitpicky, but the achievement audience tends to be very OCD by the very nature of the activity and this scenario would feel like the system is punishing you for adopting hardcore after you have some relevant softcore progression, which shouldn't happen.

All of that being said, I wouldn't be opposed of just showing totals like you proposed, but the hardcore inclined would feel like this devalues their hardcore numbers somewhat...

**Reply by Hexadigital** [reply]
I agree that a hardcore mastery supersedes a softcore mastery, but the majority of players are softcore players that have no interest in switching to hardcore. We shouldn't try to incentivize switching to hardcore more than the leaderboards and competitive aspects already do, but rather acknowledge that players can play in either mode and that there's nothing wrong with that.

What I meant by overemphasizing hardcore is that the base proposal would add a specific zero stat on softcore-only players' profiles, and that zero would stay there forever. If you look at the profile of someone who only plays softcore, like for example [Totempanic](https://retroachievements.org/user/Totempanic), you can see there's nothing on their profile relating to hardcore stats - or in other words, their profile matches their activity/playstyle.

It would be interesting to run a site-wide poll and get more feedback, but I don't think hardcore-only players would mind if it changed to just showing a total. It'd be pretty clear how they play based off their other stats and badge borders, but having the section labelled "Progression Status" show progression progress rather than progression progress split by mode would be more accurate and wouldn't lead to complaints, I believe.

**Comment by rayden-61** [comment]
Actually it does show a 0, just not on the progression bar. The Beaten
number for them should show 0 currently, since it currently only counts
hardcore (the one that counts softcore is called "Beaten (softcore)".

However, there is an inconsistency indeed, as this is only shown to the
player themselves on their "Your Completion Progress" page, but not on the
progression bar.

In any case, either of our suggestions is better than the current solution,
so I'm looking forward for any change in this regard.

Em qua., 22 de mai. de 2024, 15:00, Hexa ***@***.***>
escreveu:
